internal/js/modules/k6/browser/browser/frame_locator_mapping.go

```diff
@@ -9,9 +9,8 @@ import (
 func mapFrameLocator(vu moduleVU, fl *common.FrameLocator) mapping {
 	rt := vu.Runtime()
 	return mapping{
-		"locator": func(selector string) *sobek.Object {
-			ml := mapLocator(vu, fl.Locator(selector))
-			return rt.ToValue(ml).ToObject(rt)
+		"locator": func(selector string, opts sobek.Value) mapping {
+			return mapLocator(vu, fl.Locator(selector, parseLocatorOptions(rt, opts)))
```

Comment:
Can we really do this? There are a lot of places in the codebase where we return `rt.ToValue(ml).ToObject(rt)`, what does it do exactly and why can we remove it here?

Reply:
Yep :) Sobek automatically converts the returned `mapping`. That's what we do when we return a `mapping` in other places of the mapping layer. Turning it into an object is unnecessary because the mapping layer does that here:

https://github.com/grafana/k6/blob/3dfe1259311f3e77fd20429a65150055a005e3c9/internal/js/modules/k6/browser/browser/mapping.go#L24-L37

Reply:
Think of it like: `mapBrowser` returns a `mapping`, which in turn returns another `mapping` when one of `Browser` methods like `context` is called:

https://github.com/grafana/k6/blob/3dfe1259311f3e77fd20429a65150055a005e3c9/internal/js/modules/k6/browser/browser/browser_mapping.go#L17-L22

So, returning a `mapping` from `mapLocator` is the same.

Reply:
I do agree that we should just be letting Sobek convert from a Go [mapping](https://github.com/grafana/k6/blob/master/internal/js/modules/k6/browser/browser/mapping.go#L18) to a Sobek object. I think that's what is happening in the case of `return mapLocator(vu, fl.Locator(selector, parseLocatorOptions(rt, opts)))` as Inanc has mentioned.

I think `mapBrowserToSobek` was used as an example of what Sobek is doing under the hood when we return a `mapping` back to the Sobek runtime, internally it is converting the Go `map` to a `sobek.Object`.

Side note: I did a quick experiment with `mapBrowserToSobek`, i don't think we need to convert to a `sobek.Object`, we should just be able to work with the `mapping` i.e. `map[string]any`.